.github/workflows/main.yml

```diff
@@ -0,0 +1,247 @@
+name: Notepads CI/CD Pipeline
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  ci:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        configuration: [ Debug, Release ]
+    outputs:
+      new_version: ${{ steps.get_assembly_version.outputs.version_num }}
+      new_version_tag: ${{ steps.get_assembly_version.outputs.version_tag }}
+      latest_tag: ${{ steps.get_latest_tag.outputs.tag }}
+      is_github_release: ${{ steps.step_conditionals_handler.outputs.is_github_release }}
+    env:
+      SOLUTION_NAME: src\Notepads.sln
+      CONFIGURATION: ${{ matrix.configuration }}
+    steps:
+      - name: Steps' conditionals handler
+        id: step_conditionals_handler
+        shell: pwsh
+        run: |
+          $IS_GITHUB_RELEASE = 'false'
+          $IS_NOT_PR = 'true'
+          if ( ($env:GITHUB_EVENT_NAME -ceq 'push') -and ($env:GITHUB_REF -ceq 'refs/heads/master') -and ($env:BUILD_CONFIGURATION -ceq 'Release') ) {
```

Comment:
Public releases are built with `Production` configuration, so replace `Release` with `Production`. Or if this action will only run with  `Production` configuration then no need to do this check.